### PR TITLE
Mes 334 all on one common

### PR DESCRIPTION
**Pull request by ataylorBJSS**
- redux store, integrated into all-in-one-3
- undo functionality
- eco and eta buttons
- integrated insomnia and lock orientation

NOTE: this does not yet include underlining of last fault recorded and full logic for restricting back-marking

### CODE REVIEW

File: src/components/button-element/button-element.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  hasSerious: boolean = false;
+  hasDangerous: boolean = false;
+
+  constructor(private faultStore: FaultStoreProvider,
+    private ngRedux: NgRedux<IFaultCounter>,
+    public modalCtrl: ModalController) {
+
+    // this might not be the most performant way to do things,
+    // as it checks all the sections
+    this.ngRedux.select(state => Object.keys(state.faults)
```
Comment (danielchildsbjss) [inline src/components/button-element/button-element.ts]: Can we get this from FaultStoreProvider? Do we want the components to know about redux?

File: src/components/totals/totals.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  @select(['totals', 'dangerous']) readonly dangerousMarks$: Observable<number>;
+  @select(['totals', 'serious']) readonly seriousMarks$: Observable<number>;
+
+  constructor(private faultStore: FaultStoreProvider,
+    private ngRedux: NgRedux<IFaultCounter>) {
+
+  }
+
+  undoMark() {
+    const currentState = this.ngRedux.getState();
```
Comment (danielchildsbjss) [inline src/components/totals/totals.ts]: Can we get this from the FaultStoreProvider too?

File: src/components/totals/totals.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * See https://angular.io/api/core/Component for more info on Angular
+ * Components.
+ */
+@Component({
+  selector: 'totals',
+  templateUrl: 'totals.html'
+})
+export class TotalsComponent {
+
+  @select(['totals', 'fault']) readonly faultMarks$: Observable<number>;
```
Comment (danielchildsbjss) [inline src/components/totals/totals.ts]: Are we using $ to indicate an observable?

Comment (ataylorBJSS) [reply src/components/totals/totals.ts]: yes, It seems to be best practice in all the examples and docs I've read

File: src/components/button-element/button-element.ts (hunk 1 above)
Comment (ataylorBJSS) [reply src/components/button-element/button-element.ts]: good idea. I'll note it down for the next iteration. 👍

**Pull request merged**